### ISSUE TITLE
Allow selecting EurKEY layout in installer

### ISSUE THREAD

**Issue by xeruf** [calamares/calamares#2078]
Keycode `eu` - it is available after installation, but seemingly not in Calamares, see also https://github.com/manjaro-sway/manjaro-sway/issues/372

**Comment by boredland** [comment]
added a pr to add eurkey to the [grub menu](https://gitlab.manjaro.org/artwork/branding/grub-theme/-/merge_requests/11).

**Comment by demmm** [comment]
Since Calamares reads/uses base.lst to create the layout list:
https://github.com/calamares/calamares/blob/calamares/src/modules/keyboard/keyboardwidget/keyboardglobal.cpp#L25
it really is easiest to adjust this in the xkeyboard-config package, simple patch will add the EurKEY to the options:
```
+++ /usr/share/X11/xkb/rules/base.lst	2022-12-04 17:30:01.670795857 +0100
@@ -219,6 +219,7 @@
   dk              Danish
   nl              Dutch
   bt              Dzongkha
+  eu              EurKEY (US)
   ee              Estonian
   ir              Persian
   iq              Iraqi
```
![Screenshot_20221204_174005](https://user-images.githubusercontent.com/5360485/205503735-3977bd29-76e7-4deb-984e-ad55d013eaf4.png)

**Comment by boredland** [comment]
I don't fully understand how that repo works, but eurkey seems to be inside an ["extras" configuration](https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config/-/blob/master/rules/base.extras.xml#L1593).
From there it results in `/usr/share/X11/xkb/rules/base.extras.xml` during build.

**Comment by boredland** [comment]
[gave moving a try](https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config/-/merge_requests/449), but honestly don't know if that is how it is done.

**Comment by demmm** [comment]
Yes, it is, but since the Calamares code only reads base.lst, I tested with adding that one line to base.lst, see the image as prove that it works.

**Comment by boredland** [comment]
My upstream pr would do that, but as I am not sure what other consequences that move has, I am curious for their feedback.

**Comment by boredland** [comment]
Feedback I received [here](https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config/-/merge_requests/449#note_1672694):

> The need for extras is mostly gone now that we support custom keymaps since 99% of the extras keyboards are now better off as custom user-defined keyboard installed locally.

I don't know what that means for this endeavor.

**Comment by demmm** [comment]
Looks like either except upstream won't merge the patch, so you will have to patch locally, or adjust Calamares so it will read base-extra.xml too.

**Comment by boredland** [comment]
I would have preferred a solution that helps spreading eurkey, but honestly I don't know if my very rusty cpp skills are up for the task of adding a parser for that xml stuff.

**Comment by adriaandegroot** [comment]
The loading-of-keyboard-layouts happens in `src/modules/keyboard/keyboardwidget/keyboardglobal.cpp`, functions `parseKeyboardLayouts()` and `parseKeyboardModels()`. Those are kind of crufty things that read the `rules.lst` file. It would be fairly straightforward to make them read something **else** (e.g. the XML files). As it stands it's weird to parse the `XKB_FILE` twice and that should probably be cached somehow. Anyway, reading XML is much more structured than reading legacy flat text files, so it might be easier to do overall -- and once you can read one XML file, reading two of them and merging the results isn't a big exercise either.
